lib/oban/queue/executor.ex

```diff
@@ -147,9 +147,14 @@ defmodule Oban.Queue.Executor do
 
   defp perform_inline(%{safe: true} = exec) do
     perform_inline(%{exec | safe: false})
+  rescue
+    error ->
+      %{exec | state: :failure, error: error, stacktrace: __STACKTRACE__}
```

Comment:
I am correct that if the worker throws an exception directly (function mismatch, etc), then it'll just pass it on raw, not wrapped in `PerformError.exception`?

Reply:
That's right. This won't wrap a raised exception. Though the way it will wrap an exception returned in an error tuple, which may not be desirable.

Reply:
i.e if I return `{:error, %{something: :strange, but: :important}}` from my worker, I will lose that structured data and instead just get the exception w/ `%{something: :strange, but:important}` encoded in the message?

Reply:
No, you'll get the raw value you return. Only the formatted error stored in `errors` and the value passed to telemetry.

Reply:
> No, you'll get the raw value you return.

Where? The telemetry error handler only gets the wrapped error, right? I'm not sure where else that error would surface.

Reply:
I meant that the value is still returned from perform/1, which would only apply in testing. I see what you mean now.

The exception struct itself could have a `reason` field added that stored the original value. What do you think?

Reply:
100%. I don't see why you'd want to hide that structured data when it's easy to pass on. Maybe something like `worker_response` to make it explicit? And have it be the actual `{:error, :not_found}` `etc` so there's less "magic"?

Reply:
@coladarci I ended up using a `reason` field for consistency between the various Oban exceptions and it uses the exact value that was returned. Meaning it uses `{:error, :not_found}` or `{:discard, :some_reason}` for `PerformError`, and the value returned from caught exceptions for `CrashError`.